ui/src/actions/sampleChanger.js

```diff
@@ -58,175 +66,85 @@ export function selectDrop(drop_index) {
 
 export function refresh() {
   return (dispatch) => {
-    fetch('mxcube/api/v0.1/sample_changer/contents', {
-      method: 'GET',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      credentials: 'include',
-    }).then((response) => {
-      if (response.status >= 400) {
-        throw new Error('Error refreshing sample changer contents');
-      }
-
-      response.json().then((contents) => {
-        dispatch(setContents(contents));
-      });
+    fetchSampleChangerContents().then((json) => {
+      dispatch(setContents(json));
     });
 
-    fetch('mxcube/api/v0.1/sample_changer/loaded_sample', {
-      method: 'GET',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      credentials: 'include',
-    }).then((response) => {
-      if (response.status >= 400) {
-        throw new Error('Error refreshing sample changer contents');
```

Comment:
In case this hasn't been clear so far, when the error is just thrown like this (instead of shown in the UI via `showErrorPanel`), I've tended to simply remove the `throw` statement, since `wretch` already throws out of the box when it receives an error response.

Obviously we'll have to implement more robust error management at some point.